src/mongocxx/search_index_model.cpp

```diff
@@ -0,0 +1,56 @@
+#include <bsoncxx/stdx/make_unique.hpp>
+#include <mongocxx/private/search_index_model.hh>
+#include <mongocxx/search_index_model.hpp>
+
+#include <mongocxx/config/private/prelude.hh>
+
+namespace mongocxx {
+MONGOCXX_INLINE_NAMESPACE_BEGIN
+
+search_index_model::search_index_model(bsoncxx::document::view_or_value definition)
+    : _impl{bsoncxx::stdx::make_unique<impl>(definition)} {}
+search_index_model::search_index_model(bsoncxx::string::view_or_value name,
+                                       bsoncxx::document::view_or_value definition)
+    : _impl{bsoncxx::stdx::make_unique<impl>(name, definition)} {}
+
+search_index_model::search_index_model(search_index_model&&) noexcept = default;
+
+search_index_model& search_index_model::operator=(search_index_model&&) = default;
+
+search_index_model::search_index_model(const search_index_model& other)
+    : _impl(bsoncxx::stdx::make_unique<impl>(*other._impl)) {}
```

Comment:
```suggestion
    : _impl(bsoncxx::stdx::make_unique<impl>(other._get_impl())) {}
```
Ensure the not-null invariant is satisfied.